testdata/baselines/reference/submodule/compiler/nestedObjectRest.js.diff

```diff
@@ -1,27 +1,18 @@
 --- old.nestedObjectRest.js
 +++ new.nestedObjectRest.js
-@@= skipped -7, +7 lines =@@
- for ([{ ...y }] of [[{ abc: 1 }]]) ;
-
- //// [nestedObjectRest.js]
--var __rest = (this && this.__rest) || function (s, e) {
--    var t = {};
--    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
--        t[p] = s[p];
--    if (s != null && typeof Object.getOwnPropertySymbols === "function")
--        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
--            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
--                t[p[i]] = s[p[i]];
--        }
--    return t;
--};
+@@= skipped -18, +18 lines =@@
+         }
+     return t;
+ };
 -var _a, _b;
++var _a, _b, _c;
  // https://github.com/microsoft/TypeScript/issues/43400
  var x, y;
 -[_a] = [{ abc: 1 }], x = __rest(_a, []);
 -for (let _c of [[{ abc: 1 }]]) {
 -    [_b] = _c, y = __rest(_b, []);
-+[{ ...x }] = [{ abc: 1 }];
-+for ([{ ...y }] of [[{ abc: 1 }]])
++_a = [{ abc: 1 }], [_b] = _a, x = __rest(_b, []);
```

Comment:
In strada, there's a bunch of logic to allow sometimes inlining the temporary used as the first argument of the `__rest` call if it's not used outside the rest expression and doesn't contain anything sensitive to execution order... but it's a fair chunk of logic that doesn't map cleanly to the new, more structured visitor/transformer shape (in strada there's actually a separate visitor for value-unused expressions which enables this simplification) for only a handful of tests cleaner output. I could work on adding that in if it's important, but it hardly seemed worth it to me.